bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/basic/MiIoBasicChannel.java

```diff
@@ -25,102 +29,111 @@
  *
  * @author Marcel Verpaalen - Initial contribution
  */
+@NonNullByDefault
 public class MiIoBasicChannel {
 
     @SerializedName("property")
     @Expose
-    private String property;
+    private @Nullable String property;
     @SerializedName("friendlyName")
     @Expose
-    private String friendlyName;
+    private @Nullable String friendlyName;
     @SerializedName("channel")
     @Expose
-    private String channel;
+    private @Nullable String channel;
     @SerializedName("channelType")
     @Expose
-    private String channelType;
+    private @Nullable String channelType;
     @SerializedName("type")
     @Expose
-    private String type;
+    private @Nullable String type;
     @SerializedName("refresh")
     @Expose
-    private Boolean refresh;
+    private @Nullable Boolean refresh;
     @SerializedName("transformation")
     @Expose
-    private String transfortmation;
+    private @Nullable String transfortmation;
     @SerializedName("ChannelGroup")
     @Expose
-    private String channelGroup;
+    private @Nullable String channelGroup;
     @SerializedName("actions")
     @Expose
-    private List<MiIoDeviceAction> miIoDeviceActions = new ArrayList<MiIoDeviceAction>();
+    private @Nullable List<MiIoDeviceAction> miIoDeviceActions = new ArrayList<MiIoDeviceAction>();
 
     public String getProperty() {
-        return property;
+        final String property = this.property;
+        return (property != null) ? property : "";
     }
 
     public void setProperty(String property) {
         this.property = property;
     }
 
     public String getFriendlyName() {
-        return type == null || friendlyName.isEmpty() ? channel : friendlyName;
+        final String fn = friendlyName;
+        return (fn == null || type == null || fn.isEmpty()) ? getChannel() : fn;
     }
 
     public void setFriendlyName(String friendlyName) {
         this.friendlyName = friendlyName;
     }
 
     public String getChannel() {
-        return channel;
+        final @Nullable String channel = this.channel;
+        return channel != null ? channel : "";
     }
 
     public void setChannel(String channel) {
         this.channel = channel;
     }
 
     public String getChannelType() {
-        return channelType == null || channelType.isEmpty() ? BINDING_ID + ":" + channel
-                : (channelType.startsWith("system") ? channelType : BINDING_ID + ":" + channelType);
+        final @Nullable String ct = channelType;
+        return ct == null || ct.isEmpty() ? BINDING_ID + ":" + channel
+                : (ct.startsWith("system") ? ct : BINDING_ID + ":" + ct);
```

Comment:
too many conditional operators and `":"` mixed with `:`. Can you make this more readable?

Reply:
I improved this now. I think it is more clear